### PR TITLE
Update book

### LINKED ISSUE #1009: Update docs mentioning that some metrics are supported for Java

### PR DESCRIPTION
**Pull request by Luni-4**
This PR updates supported languages and adds new metrics to the documentation. It fixes #1009

### CODE REVIEW

File: rust-code-analysis-book/src/languages.md [hunk 1]
```diff
@@ -4,15 +4,12 @@ This is the list of programming languages parsed by
 **rust-code-analysis**.
 
 - [x] C++
-- [ ] C#
```
Comment (marco-c) [inline rust-code-analysis-book/src/languages.md:7]: Why not keeping C# and Go in the list?

File: rust-code-analysis-book/src/metrics.md [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
+- **CC**: it calculates _Cyclomatic_ complexity examining the
   control flow of a program.
-- **SLOC**: it counts the number of lines in a source file.
-- **PLOC**: it counts the number of physical lines (instructions) contained in
-  a source file.
-- **LLOC**: it counts the number of logical lines (statements) contained in
-  a source file.
 - **CLOC**: it counts the number of comments in a source file.
-- **BLANK**: it counts the number of blank lines in a source file.
+- **COGNITIVE**: it calculates _Cognitive_ complexity examining whether a code
```
Comment (marco-c) [inline rust-code-analysis-book/src/metrics.md:11]: I'd add a link to the definition of cognitive complexity, and change the description to be:
> it calculates the _Cognitive_ complexity, measuring how complex it is to understand a unit of code.

File: rust-code-analysis-book/src/languages.md (hunk 1 above)
Comment (Luni-4) [reply rust-code-analysis-book/src/languages.md:7]: We do not support them right now

Comment (marco-c) [reply rust-code-analysis-book/src/languages.md:7]: We don't support Kotlin either, but you added it. I think it's good to list the ones we would like to support without the X

Comment (Luni-4) [reply rust-code-analysis-book/src/languages.md:7]: We still don't have metrics for Kotlin but we support it as parsed language https://github.com/mozilla/rust-code-analysis/blob/master/Cargo.toml#L32

**Review by marco-c — APPROVED** [review]

**Pull request merged**